ai/integrations/autogen/src/unitycatalog/ai/autogen/toolkit.py

```diff
@@ -15,91 +12,64 @@
+    def __init__(
+        self,
+        name: str,
+        description: str,
+        function_full_name: str,
+        client: BaseFunctionClient,
+        args_type: BaseModel,
+    ):
         """
-        Converts the Autogen Tool instance into a dictionary for the Autogen API.
+        Args:
+            name: The short tool name (what the LLM sees for `function_call`).
+            description: A brief description.
+            function_full_name: e.g. "catalog.schema.function".
+            client: A Unity Catalog client with `.execute_function(...)`.
+            args_type: A Pydantic model class representing the function's input parameters.
         """
-        return {"name": self.name, "description": self.description, "tool": self.tool}
+        super().__init__(
+            args_type=args_type,
+            return_type=str,
+            name=name,
+            description=description,
+        )
+        self._function_full_name = function_full_name
+        self._client = client
 
-    def register_function(
-        self,
-        *,
-        callers: Union[ConversableAgent, List[ConversableAgent]],
-        executors: Union[ConversableAgent, List[ConversableAgent]],
-    ) -> None:
+    async def run(self, args: BaseModel, cancellation_token: CancellationToken) -> str:
         """
-        Registers the function associated with the Autogen tool for all combinations of callers and executors.
-
-        This method registers the callable function (`fn`) with each provided executor.
-        It also updates the tool signature with each caller to ensure that the function's definition
-        is correctly reflected in the overall toolset.
-
-        Args:
-            callers (ConversableAgent or List[ConversableAgent]): The agent(s) that will call the tool.
-            executors (ConversableAgent or List[ConversableAgent]): The agent(s) that will execute the tool's function.
+        Actually call the UC function with the provided parameters.
         """
-        # Ensure callers is a list
-        if isinstance(callers, ConversableAgent):
-            callers = [callers]
-        elif not isinstance(callers, list):
-            raise TypeError(
-                "callers must be a ConversableAgent or a list of ConversableAgent instances."
-            )
-
-        # Ensure executors is a list
-        if isinstance(executors, ConversableAgent):
-            executors = [executors]
-        elif not isinstance(executors, list):
-            raise TypeError(
-                "executors must be a ConversableAgent or a list of ConversableAgent instances."
-            )
-
-        # Update tool signature for each caller
-        for caller in callers:
-            caller.update_tool_signature(self.tool, is_remove=False)
-
-        # Register function with each executor
-        for executor in executors:
-            executor.register_function({self.name: executor._wrap_function(self.fn)})
+        param_dict = args.model_dump()
```

Comment:
is pydantic version pinned?

Reply:
They only support 2.x and py 3.10 with their library refactoring